AxonDeepSeg/download_model.py

```diff
@@ -14,9 +15,17 @@ def download_model(destination = None):
         tem_destination = destination / "default_TEM_model"
         model_seg_pns_bf_destination = destination / "model_seg_pns_bf"
 
-    url_TEM_model = "https://osf.io/2hcfv/?action=download&version=5"  # URL of TEM model hosted on OSF storage with the appropriate versioning on OSF
-    url_SEM_model = "https://osf.io/sv7u2/?action=download&version=5"  # URL of SEM model hosted on OSF storage with the appropriate versioning on OSF
-    url_model_seg_pns_bf = "https://osf.io/hfeu2/?action=download&version=2" # URL of Bright Field Optical Microscopy model hosted on storage with the appropriate versioning on OSF
+    # retrieve latest release identifier from github API
+    response_TEM = requests.get("https://api.github.com/repos/axondeepseg/default-TEM-model/releases/latest")
+    folder_name_TEM_model = Path("default-TEM-model-"+str(response_TEM.json()['tag_name']))
```

Comment:
Thanks @hermancollin for creating your first PR 🎉 in ADS. 

I've a question: is it not possible to directly download the latest release without extracting the tag name? 

I was able to extract the asset zip without having  the tag name in my PR see here. https://github.com/neuropoly/axondeepseg/pull/518#issuecomment-863351415

Likewise, I was able to download the latest release for **SEM**, **TEM** and **OM** models

1. https://api.github.com/repos/axondeepseg/default-SEM-model/zipball
2.  https://api.github.com/repos/axondeepseg/default-TEM-model/zipball
3. https://api.github.com/repos/axondeepseg/model-seg-pns-bf/zipball

Reply:
Yes, I agree with you that the method I suggested looks a bit tedious. The problem with these links is that they are not compatible with the `ads_utils.download_data()` function. I played around with variations of these URLs a lot and what I found is that we always get the same problem: the repo is zipped inside another folder, so when the `download_data` function retrieves the content from the URL, we need to know the name of the highest-level folder.

The URLs you suggested give us the following:
![Screenshot_20210621_115911](https://user-images.githubusercontent.com/83031821/122792382-1f73cc80-d288-11eb-9766-82740ba19b8b.png)
As you can see, the folder name contains the first 6 characters from the commit hash, which is slightly more complicated to retrieve than the tag_name. I know this isn't ideal, but I haven't found a better way to deal with this.

I had another idea to deal with this problem. When creating the release, we can manually add the compressed file in the assets and this would be easier to retrieve directly than the default released zip file. However, this workflow is really not ideal because whenever we will want to change something in the repo, we'll have to create a new release and manually add the compressed file everytime (also the name of the added asset would need to be the same for every release to be compatible with a hardcoded asset name).

Reply:
> I had another idea to deal with this problem. When creating the release, we can manually add the compressed file in the assets and this would be easier to retrieve directly than the default released zip file. However, this workflow is really not ideal because whenever we will want to change something in the repo, we'll have to create a new release and manually add the compressed file everytime (also the name of the added asset would need to be the same for every release to be compatible with a hardcoded asset name).

I agree with you. Thats not the best way to go ahead with. However, I wish to know how does `ivadomed` deals with data and releases on GH. @jcohenadad any ideas? 

Your first solution makes sense and I'm favour of it.

Reply:
> I've a question: is it not possible to directly download the latest release without extracting the tag name?

We definitely want to keep the tag name in the URL, in order to ensure cross-compatibility. Eg: a user would like to analyse a dataset using a version of ADS that is 2 years old: the _latest_ models would be different from that one 2 years ago, and the users would not be able to reproduce results.

The way we do it in SCT is by pointing to the tagged URL (example [here](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/scripts/sct_download_data.py#L53)), downloading/unzipping in a temporary folder, and then copy the model/data in a folder that always has the same name (eg: model-tem). 

> I had another idea to deal with this problem. When creating the release, we can manually add the compressed file in the assets and this would be easier to retrieve directly than the default released zip file.

Every idea that has the name "manually" in it is a bad idea 😉

Reply:
> The way we do it in SCT is by pointing to the tagged URL (example [here](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/scripts/sct_download_data.py#L53)), downloading/unzipping in a temporary folder, and then copy the model/data in a folder that always has the same name (eg: model-tem).

Should we put the links in the `config.py` script, as suggested by @vasudev-sharma or should we put them directly in the `download_models.py` script (like the *SCT way*)?

Reply:
hum... i find it less cumbersome from a developer standpoint to put the script URL where it is used (ie: in download_models.py).

Reply:
I also prefer to keep the links in `download_model.py`, this is where I would intuitively look for them.